richtextfx/src/main/java/org/fxmisc/richtext/skin/TextExt.java

```diff
@@ -0,0 +1,95 @@
+package org.fxmisc.richtext.skin;
+
+import com.sun.javafx.css.converters.PaintConverter;
+import javafx.beans.property.ObjectProperty;
+import javafx.css.CssMetaData;
+import javafx.css.Styleable;
+import javafx.css.StyleableObjectProperty;
+import javafx.css.StyleableProperty;
+import javafx.scene.paint.Color;
+import javafx.scene.paint.Paint;
+import javafx.scene.text.Text;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TextExt extends Text {
+
+    private ObjectProperty<Paint[]> backgroundColor = null;
```

Comment:
A nifty customisation would be controlling line style and background offset
from/around text. This will enable highlighting text in different ways. Not
sure how hard it is to implement though.

In my mind, one of the things to consider is how the background will work
when making text selection. For example, if selection overlaps an already
coloured text background?

I look forward to plying with your addition :)


On Aug 17, 2015 3:58 AM, "Mewes Kochheim" <notifications@github.com> wrote:

> In richtextfx/src/main/java/org/fxmisc/richtext/skin/TextExt.java
> <https://github.com/TomasMikula/RichTextFX/pull/166#discussion_r37176819>:
>
> > +import javafx.beans.property.ObjectProperty;
> > +import javafx.css.CssMetaData;
> > +import javafx.css.Styleable;
> > +import javafx.css.StyleableObjectProperty;
> > +import javafx.css.StyleableProperty;
> > +import javafx.scene.paint.Color;
> > +import javafx.scene.paint.Paint;
> > +import javafx.scene.text.Text;
> > +
> > +import java.lang.reflect.Field;
> > +import java.util.ArrayList;
> > +import java.util.List;
> > +
> > +public class TextExt extends Text {
> > +
> > +    private ObjectProperty<Paint[]> backgroundColor = null;
>
> I just changed the code based on your last comment. I will test the latest
> version tonight a bit more.
>
> My next pull request will be about line styles (think red background for
> lines with break point or or yellow background for current selected line).
> Any suggestions to make it more likely to get merged upstream?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/TomasMikula/RichTextFX/pull/166/files#r37176819>.
>